benchmarks/tool/compile_benchmarks.dart

```diff
@@ -0,0 +1,184 @@
+    for (final target in targets) {
+      commands.add(target.compileArgs(sourceFile));
+    }
+  }
+
+  final pool = Pool(jobs);
+
+  final stream = pool.forEach<List<String>, ProcessResult>(commands,
+      (List<String> command) {
+    print(command.join(' '));
+    return Process.run(command[0], command.sublist(1));
+  });
+
+  await for (final processResult in stream) {
+    final exitCode = processResult.exitCode;
+    if (exitCode != 0) {
+      print('Process exited with $exitCode');
+      print(
+          'Process stdout ---------------------------------------------------');
+      print(processResult.stdout);
+      print(
+          'Process stderr ---------------------------------------------------');
+      print(processResult.stderr);
+      print(
+          '------------------------------------------------------------------');
+      exit(1);
+    }
+  }
+
+  await pool.done;
+}
+
+class Target {
+  final String _name;
+  final List<String> Function(String) _processArgs;
+
+  const Target(this._name, this._processArgs);
+
+  @override
+  String toString() {
+    return 'Target($_name)';
+  }
+
+  List<String> compileArgs(String sourceFile) {
+    return _processArgs(sourceFile);
+  }
+}
+
+const aotTarget = Target('aot', aotProcessArgs);
+const exeTarget = Target('exe', exeProcessArgs);
+const jitTarget = Target('jit', jitProcessArgs);
+const jsTarget = Target('js', jsProcessArgs);
+const jsProductionTarget = Target('js-production', jsProductionProcessArgs);
+
+List<String> aotProcessArgs(String sourceFile) {
+  final baseName = path.basename(sourceFile);
+  final baseNameNoExt = path.withoutExtension(baseName);
+  return [
+    'dart',
+    'compile',
+    'aot-snapshot',
+    sourceFile,
+    '-o',
+    'out/$baseNameNoExt.aot'
+  ];
+}
+
+List<String> exeProcessArgs(String sourceFile) {
+  final baseName = path.basename(sourceFile);
+  final baseNameNoExt = path.withoutExtension(baseName);
+  return ['dart', 'compile', 'exe', sourceFile, '-o', 'out/$baseNameNoExt.exe'];
+}
+
+List<String> jitProcessArgs(String sourceFile) {
+  final baseName = path.basename(sourceFile);
+  final baseNameNoExt = path.withoutExtension(baseName);
+  return [
+    'dart',
+    '--snapshot-kind=kernel',
+    '--snapshot=out/$baseNameNoExt.dill',
+    sourceFile
+  ];
+}
+
+List<String> jsProcessArgs(String sourceFile) {
+  final baseName = path.basename(sourceFile);
+  final baseNameNoExt = path.withoutExtension(baseName);
+  return ['dart', 'compile', 'js', sourceFile, '-o', 'out/$baseNameNoExt.js'];
+}
+
+List<String> jsProductionProcessArgs(String sourceFile) {
+  // This is used in golem "...-dart2js-production" benchmarks.
+  // `--benchmarking-production` is not documented, it enables
+  // `--omit-implicit-checks` and `--trust-primitives`:
+  // https://github.com/dart-lang/sdk/blob/c48f6fea580178bd34f2d872588dcc1c79bdb01c/pkg/compiler/lib/src/options.dart#L764-L767
+  //
+  // We will probably want to use `-O4` here.
```

Comment:
--benchmarking-production is undocumented, but is meant to be whatever the web people currently consider the options one would use for production compilations of real world things (including slightly unsafe options which will change over time)

Reply:
I wanted to use `--benchmarking-production` because that's what we use in golem and I wanted to be compatible with the existing results, but since then I updated my golem CL and removed code to run old revisions. I think we can use `-O4` now. (#552)